src/Error/ExceptionRenderer.php

```diff
@@ -390,29 +390,37 @@ protected function _outputMessage(string $template): Response
             $this->controller->render($template);
 
             return $this->_shutdown();
+        } catch (MissingLayoutException | MissingPluginException $e) {
+            // Configuration error, bail out of everything
         } catch (MissingTemplateException $e) {
-            $attributes = $e->getAttributes();
-            if (
-                $e instanceof MissingLayoutException ||
-                (
-                    isset($attributes['file']) &&
-                    strpos($attributes['file'], 'error500') !== false
-                )
-            ) {
-                return $this->_outputMessageSafe('error500');
+            $request = $this->controller->getRequest();
+
+            // Bail out of one configuration at a time trying to find template
+            if ($request->getParam('_ext')) {
+                $this->controller->setRequest($request->withParam('_ext', null));
+
+                return $this->_outputMessage($template);
             }
 
-            return $this->_outputMessage('error500');
-        } catch (MissingPluginException $e) {
-            $attributes = $e->getAttributes();
-            if (isset($attributes['plugin']) && $attributes['plugin'] === $this->controller->getPlugin()) {
+            if ($request->getParam('prefix')) {
+                $this->controller->setRequest($request->withParam('prefix', null));
+
+                return $this->_outputMessage($template);
+            }
+
+            if ($this->controller->getPlugin()) {
                 $this->controller->setPlugin(null);
+                $this->controller->viewBuilder()->setPlugin(null);
+
+                return $this->_outputMessage($template);
             }
 
-            return $this->_outputMessageSafe('error500');
-        } catch (Throwable $e) {
-            return $this->_outputMessageSafe('error500');
+            if ($template !== 'error500') {
+                return $this->_outputMessage('error500');
```

Comment:
:crossed_fingers: that these 10 ish additional stack frames don't blow the limit :smile:

Reply:
I thought of that too. This re-uses the original functions as-is, but will see if I have an idea how to iterate without recursion.